addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -1494,9 +1497,28 @@ options.registry.WebsiteFormFieldModel = DisableOverlayButtonOption.extend({
 // Disable delete button for model required fields
 options.registry.WebsiteFormFieldRequired = DisableOverlayButtonOption.extend({
     start: function () {
-        this.disableButton('remove', _t('This field is mandatory for this Action. You cannot remove it.'));
+        this.disableButton("remove", _t(
+            "This field is mandatory for this action. You cannot remove it. Try hiding it with the"
+            + " 'Visibility' option instead and add it a default value."
+        ));
         return this._super.apply(this, arguments);
-    }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    async _renderCustomXML(uiFragment) {
+        const fieldName = this.$target[0]
+            .querySelector("input.s_website_form_input").getAttribute("name");
+        const spanEl = document.createElement("span");
+        spanEl.innerText = _.str.sprintf(
+            _t("The field '%s' is mandatory for the action '%s'."), fieldName, currentActionName);
+        uiFragment.querySelector("we-alert").appendChild(spanEl);
```

Comment:
Not sure if it is a good idea but can't this be done directly in the `start` ? You could then use `this.$el[0]` instead of `uiFragment` to access the options block :wink:  (I tried and it works)
I wonder because from what I saw, `renderCustomXML` is called only once so it could be directly made at the start.


Also, instead of creating a `span` element here in JS, what do you think about creating a XML template for the `we-alert` with the `span` with the text already inside of it ? The `fieldName` and `currentActionName` would be parameters set when rendering the template in the start (with the call to `qweb.render`) and used with `t-out` in the `span`.
Something like this: 
```js
const alertEl = qweb.render(templateName, {field: fieldName, action: currentActionName})
this.$el[0].insertAdjacentHTML("afterbegin", alertEl);
``` 
I'm not sure if it is better to render a qweb or to do like you did :thinking: Just a suggestion :wink:

Reply:
> Not sure if it is a good idea but can't this be done directly in the start ?

I think it's a bad idea... I don't see existing code that does it. I think the good practice is to do it in `renderCustomXML`.

> what do you think about creating a XML template

If it was a "complex" HTML structure to render I would clearly agree with you but here as it is just a simple `span` with text inside to render, I wonder if it's not overkill to do it :thinking: